### DISCUSSION TITLE
Tailwind config for package being used in multiple apps

### DISCUSSION THREAD

**Discussion by karanmidha** [tailwindlabs/tailwindcss#13535] [Help]
As a part of my organization's design system, I have a package called "core" from which I am exporting all the CSS. The apps have it as a dependency and it gets installed as a `node_module` in their application.

I recently added tailwindcss to my "core" package but tailwind classes won't apply to applications using my package since they cannot specify the `content` path. My goal export tailwind from my package so that the application will have to do minimum configuration on their end and they can start using tailwind.

```
module.exports = {
  content: [], -> This resides in my "core" package and apps cannot modify it
  theme: {
    extend: {}
  },
  plugins: []
};
```

One option I have is to use `safelist` property in the `tailwind.config.js` file and add all the CSS that I need but I don't want to do that. I want to have a way where the end application can use any tailwind class they want to and also benefit from the JIT compiler/purge feature of tailwind, so that their bundle doesn't include all the tailwind CSS.

How can the applications using my package specify the `content` path dynamically?

PS - I am using `rollup` if that makes any difference.

Any help is appreciated, thanks!

**Comment by alex-krasikau** [comment]
Hey @karanmidha!

It's fine to reference the `node_modules` package in the application's Tailwind config. If your `core` package has a theme configuration, you can import and use it as a module.

```js
// app/tailwind.config.js

const { theme } = require("@org/base")

module.exports = {
  content: [
    "./node_modules/@org/code/src/**/*.{js,ts,jsx,tsx}",
    "./src/**/*.{js,ts,jsx,tsx}",
  ],
  theme: {
    extend: {
      ...theme
    },
  },
  plugins: []
};
```

**Reply by karanmidha** [reply]
Thanks for answering my problem @alex-krasikau. 

My ask is, is it possible for the applications using my "core" package to not install tailwind at their end and still use tailwind capabilities since they are importing my package and my package has tailwind installed? The problem I am facing is that when they install my package "core", the "core" doesn't have a value for `content` and hence, no CSS is being returned. 

I am looking for a way that the application can modify the config of the "core" package(node module in their case) and the package gets built again. I don't know if this thing is even possible.

**Reply by alex-krasikau** [reply]
Got it. You want to provide a component library styled with Tailwind CSS. The application installs the 'core' package, but it may use other styling options, such as CSS modules. The application should be able to customize theme settings like colors. Is that correct?

**Reply by karanmidha** [reply]
You got me. The package "core" is just exporting some styles and along with those styles, now, we want to export tailwind as well. The intent is that the consumer application can use the utility classes without having to setup tailwind(or minimal setup) at their end. 

Customizing theme is not a scenario as of now as we have our own set of themes being used across the organization.

**Reply by alex-krasikau** [reply]
It appears that you still need to set up Tailwind and PostCSS in the project. Then import the full configuration from the `core` module. Uncertain how it will work with editor tooling like autocomplete.
